### PR TITLE
PVST feature support

### PR DESCRIPTION
**Pull request by divyachandralekha**
Added new stpmgr for STP config handling
Added changes in orchagent for STP programming via SAI APIs

What I did

Added new stpmgr under cfgmgr for handling STP config notifications
Added STP changes in stporch component to handle STP operations (STP Instance creation/deletion, STP Port Add/Update/Del to STP instance) via SAI

Why I did it
PVST Feature support

How I Verified it
Added UT cases for stporch changes

### CODE REVIEW

**Comment by divyachandralekha** [comment]
This PR related to  PR #3316

**Comment by divyachandralekha** [comment]
@ridahanif96 @wajahatrazi @muhammadalihussnain @bendrapubalareddy @praveenraja1  @prsunny @srj102 , Please review and approve the changes

**Review by wajahatrazi — APPROVED** [review]
LGTM

**Review by srj102 — APPROVED** [review]
Same as #3316

File: orchagent/stporch.cpp [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+    m_stpTable = unique_ptr<Table>(new Table(stateDb, STATE_STP_TABLE_NAME));
+    
+    vector<sai_attribute_t> attrs;
+    attr.id = SAI_SWITCH_ATTR_DEFAULT_STP_INST_ID;
+    attrs.push_back(attr);
+    
+    status = sai_switch_api->get_switch_attribute(gSwitchId, (uint32_t)attrs.size(), attrs.data());
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        throw runtime_error("StpOrch initialization failure");
```
Comment (oleksandrivantsiv) [inline orchagent/stporch.cpp:33]: @divyachandralekha why is an exception thrown here? Dedending on the vendor and the device on which SONiC is running SAI may return `SAI_STATUS_NOT_SUPPORTED` or `SAI_STATUS_NOT_IMPLEMENTED`. The StpOrch should handle those return codes and should not prevent Orchagent from starting.

@prgeor FYI. This breaks SONiC on the Smart Switch DPU.

Comment (divyachandralekha) [reply orchagent/stporch.cpp:33]: Fixed and pushed the changes for review. 
https://github.com/sonic-net/sonic-swss/pull/3449

**Pull request merged**